mm2src/coins/eth/eth_swap_v2.rs

```diff
@@ -96,14 +129,327 @@ impl EthCoin {
+            dex_fee,
+            payment_amount,
+            maker_address,
+            taker_secret_hash: try_tx_s!(taker_secret_hash.try_into()),
+            maker_secret_hash: try_tx_s!(maker_secret_hash.try_into()),
+            payment_time_lock,
+            token_address,
+        };
+        let data = try_tx_s!(self.prepare_taker_refund_payment_timelock_data(args).await);
+
+        self.sign_and_send_transaction(
+            0.into(),
+            Action::Call(taker_swap_v2_contract),
+            data,
+            U256::from(gas_limit),
+        )
+        .compat()
+        .await
+    }
+
+    pub(crate) async fn refund_taker_funding_secret_impl(
+        &self,
+        args: RefundFundingSecretArgs<'_, Self>,
+    ) -> Result<SignedEthTx, TransactionErr> {
+        let taker_swap_v2_contract = self
+            .swap_v2_contracts
+            .as_ref()
+            .map(|contracts| contracts.taker_swap_v2_contract)
+            .ok_or_else(|| TransactionErr::Plain(ERRL!("Expected swap_v2_contracts to be Some, but found None")))?;
+        let taker_secret = try_tx_s!(args.taker_secret.try_into());
+        let maker_secret_hash = try_tx_s!(args.maker_secret_hash.try_into());
+        let dex_fee = try_tx_s!(wei_from_big_decimal(
+            &args.dex_fee.fee_amount().to_decimal(),
+            self.decimals
+        ));
+        let payment_amount = try_tx_s!(wei_from_big_decimal(
+            &(args.trading_amount + args.premium_amount),
+            self.decimals
+        ));
+        let maker_address = public_to_address(args.maker_pubkey);
+        let payment_time_lock: u32 = try_tx_s!(args.payment_time_lock.try_into());
+        let (token_address, gas_limit) = match &self.coin_type {
+            // TODO need new consts and params for v2 calls. now it uses v1
+            EthCoinType::Eth => (Address::default(), self.gas_limit.eth_sender_refund),
+            EthCoinType::Erc20 {
+                platform: _,
+                token_addr,
+                // TODO need new consts and params for v2 calls. now it uses v1
+            } => (*token_addr, self.gas_limit.erc20_sender_refund),
+            EthCoinType::Nft { .. } => {
+                return Err(TransactionErr::ProtocolNotSupported(ERRL!(
+                    "NFT protocol is not supported for ETH and ERC20 Swaps"
+                )))
+            },
+        };
+        let refund_args = TakerRefundSecretArgs {
+            dex_fee,
+            payment_amount,
+            maker_address,
+            taker_secret,
+            maker_secret_hash,
+            payment_time_lock,
+            token_address,
+        };
+        let data = try_tx_s!(self.prepare_taker_refund_payment_secret_data(&refund_args).await);
+
+        self.sign_and_send_transaction(
+            0.into(),
+            Action::Call(taker_swap_v2_contract),
+            data,
+            U256::from(gas_limit),
+        )
+        .compat()
+        .await
+    }
```

Comment:
same as above and any other places too

Reply:
Done https://github.com/KomodoPlatform/komodo-defi-framework/pull/2169/commits/9f3164995bd463227c570ad9bbd890e3103142ec

also moved coint_type match in other function in this commit https://github.com/KomodoPlatform/komodo-defi-framework/pull/2169/commits/046dc323c72999da8d35874e2934f9c81da3a1bf